src/main/java/frc/robot/Constants.java

```diff
@@ -79,38 +80,46 @@ public static class ShooterConstants {
   }
 
   public static class TurretConstants {
-    public static final int TURRET_MOTOR_ID = 9;
+    public static final int TURRET_MOTOR_ID = 2;
     public static final int TURRET_CANcoder_ID = 17; // replace with actual CANcoder ID
-    public static final int PITCH_MOTOR_ID = 10;
+    public static final int PITCH_MOTOR_ID = 3;
     public static final int PITCH_ENCODER_ID = 1; // replace with actual ID
     public static final boolean IS_MOTOR_INVERTED = true;
+    public static final int DEFAULT_SHOT_PITCH = 40;
+    public static final int DEFAULT_SHOT_ROTATION = 0;
 
     // PID coefficients
-    // TODO: Change values when there's an actual real functional robot.
-    public static final HowdyPID TURRET_PID = new HowdyPID(0.125, 0.001, 0);
+    public static final boolean ADVANCE_LOOP = true;
+    // Position in turret position (-0.5 to 0.5) to velocity in RPM
+    public static final HowdyPID TURRET_POSITION_PID_CASCADE = new HowdyPID(200, 0., 0., 0.0, 0);
+    // Velocity in RPM to percent out
+    public static final HowdyPID TURRET_VELOCITY_PID_CASCADE =
+        new HowdyPID(0.05, 1e-8, 3e-5, 4.0, 0);
 
     public static final double TURRET_KMAXOUTPUT = 1;
     public static final double TURRET_KMINOUTPUT = -1;
 
-    public static final int TURRET_MIN_ANGLE_DEGREES = -5;
-    public static final int TURRET_MAX_ANGLE_DEGREES = 45;
+    public static final int TURRET_MIN_ANGLE_DEGREES = -90;
+    public static final int TURRET_MAX_ANGLE_DEGREES = 90;
 
-    public static final double TURRET_CONVERSION_FACTOR = 0.25;
-    public static final int TURRET_SMART_CURRENT_LIMIT = 40;
+    public static final double TURRET_CONVERSION_FACTOR = 0.18292682926829268292682926829268;
```

Comment:
This is an oddly lengthy number, but also it repeats. Could we represent it as a fraction instead?